terasoluna-tourreservation-domain/src/test/java/org/terasoluna/tourreservation/domain/service/reserve/ReserveServiceImplSecurityTest.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright (C) 2013-2015 terasoluna.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+package org.terasoluna.tourreservation.domain.service.reserve;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.terasoluna.gfw.common.date.jodatime.JodaTimeDateFactory;
+import org.terasoluna.tourreservation.domain.model.*;
+import org.terasoluna.tourreservation.domain.repository.reserve.ReserveRepository;
+import org.terasoluna.tourreservation.domain.service.userdetails.ReservationUserDetails;
+
+import javax.inject.Inject;
+
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsNull.nullValue;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.*;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration
+public class ReserveServiceImplSecurityTest {
+
+    @Inject
+    ReserveRepository mockReserveRepository;
+
+    @Inject
+    ReserveService reserveService;
+
+    @Inject
+    JodaTimeDateFactory dateFactory;
+
+    @Before
+    @After
```

Comment:
Added  `@Before` for safely.
However, `@Before` is not required. I will remove it.